File: 2488-divide-intervals-into-minimum-number-of-groups/2488-divide-intervals-into-minimum-number-of-groups.java
class Solution {
    public int minGroups(int[][] intervals) {
        List<Integer> start = new ArrayList<>();
        List<Integer> end = new ArrayList<>();

        for(int[] interval : intervals){
            start.add(interval[0]);
            end.add(interval[1]);
        }

        Collections.sort(start);
        Collections.sort(end);

        int groupCount = 0;
        int endPtr = 0;

        for(int s : start){
            if(s > end.get(endPtr)){
                endPtr++;
            } else {
                groupCount++;
            }
        }
        return groupCount;
    }
}

File: 2488-divide-intervals-into-minimum-number-of-groups/2488-divide-intervals-into-minimum-number-of-groups.py
class Solution(object):
    def minGroups(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        start = []
        end = []

        for interval in intervals:
            start.append(interval[0])
            end.append(interval[1])
        
        start.sort()
        end.sort()

        group_count = 0
        end_ptr = 0

        for s in start:
            if s > end[end_ptr]:
                end_ptr += 1
            else:
                group_count += 1
        
        return group_count
        